witness/src/Dscp/Snowdrop/PublicationValidation.hs

```diff
@@ -0,0 +1,106 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Dscp.Snowdrop.PublicationValidation
```

Comment:
I think all this functionality should be in `Validation`.

Reply:
All this module functionality, you mean? We have `AccountValidation`, why not have separate module for `PublicationValidation`?